pd/src/state.rs

```diff
@@ -59,11 +67,23 @@ INSERT INTO blobs (id, data) VALUES ('gc', $1)
         .execute(&mut dbtx)
         .await?;
 
-        // TODO: move into BaseRateData::at_genesis() ctor?
-        query!(
-            "INSERT INTO base_rates (epoch, base_reward_rate, base_exchange_rate) VALUES (0, 0, $1)",
-            1_0000_0000
-        ).execute(&mut dbtx).await?;
+        // Delegations require knowing the rates for the next epoch, so
+        // pre-populate with 0 reward => exchange rate 1 for the current
+        // (index 0) and next (index 1) epochs.
+        for epoch in [0, 1] {
+            query!(
+                "INSERT INTO base_rates (
+                epoch, 
+                base_reward_rate, 
+                base_exchange_rate
+            ) VALUES ($1, $2, $3)",
+                epoch,
+                0,
+                1_0000_0000
```

Comment:
This is written somewhat confusingly: `1_0000_0000` looks at a glance like "1 million" but it's actually "100 million". Is this intentional?

Reply:
Yep, the value is in basis points of basis points, so the digits are grouped in 10^4-sized chunks, and this is supposed to be read as "1.0000.0000"